### ISSUE TITLE
-target thumb-freestanding-none includes std.target.Arch.isWasm

### ISSUE THREAD

**Issue by markfirmware** [ziglang/zig#4715]
export fn mission1_main() void {}

    pub fn panic(message: []const u8, trace: ?*@import("std").builtin.StackTrace) noreturn {
        while (true) {}
    }

    zig build-exe -target thumb-freestanding-none --linker-script linker.ld main.zig
    llvm-objdump -x --source zig-cache/bin/main > main.asm
    grep -i wasm main.asm

    0000000000000012 std.target.Arch.isWasm:
    ; pub fn isWasm(arch: Arch) bool {
          26:	04 d3 	blo	#8 <std.target.Arch.isWasm+0x20>
          28:	ff e7 	b	#-2 <std.target.Arch.isWasm+0x18>
          30:	03 e0 	b	#6 <std.target.Arch.isWasm+0x28>
    ; .wasm32, .wasm64 => true,
          38:	ff e7 	b	#-2 <std.target.Arch.isWasm+0x28>
    00000012 l     F .text		 0000002e std.target.Arch.isWasm

**Comment by LemonBoy** [comment]
TLDR: Use `-ffunction-sections`

The compiler generates that function in case you may want to invoke it from a debugger, there's already an open ticket about being able to mark some functions as comptime-only.

**Comment by markfirmware** [comment]
Thanks. Let me think about that.

**Comment by markfirmware** [comment]
Ok, -ffunction-sections works.

**Comment by markfirmware** [comment]
@LemonBoy @andrewrk 

How do I add -ffunction-sections in build.zig?

**Comment by markfirmware** [comment]
https://github.com/ziglang/zig/pull/4746/commits/27be16543e0370f795be0a596f2668a205583e51

> @LemonBoy @andrewrk
> 
> How do I add -ffunction-sections in build.zig?

Thanks!

**Comment by frmdstryr** [comment]
Would something like a `callconv(.Comptime)` work to tag functions intended for comptime only?
